### DISCUSSION TITLE
Need to show swatches on Recently view product carousel

### DISCUSSION THREAD

**Discussion by rutulritz** [breezefront/community#66] [1. Help]
Hi,

How can I add/enable hover swatches on Recently viewed product carousel on PDP ? 

I checked and the products are rendering from breeze template of recent-products. 

Can you please help me for this?

Thanks

**Comment by vovayatsyuk** [comment]
Breeze doesn't provide a ready-to-use solution for this. How would you do this in Luma? 

Since this block is fully client-side, you need to add swatches info to this JSON data:

![image](https://user-images.githubusercontent.com/306080/221514824-001371d4-c33e-4368-83e8-c91fb28cc186.png)

To do this you need to create a plugin for [Magento\Catalog\Block\Ui\ProductViewCounter](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Catalog/Block/Ui/ProductViewCounter.php)

When the swatch data will be available in the json, you can override [recent-products.phtml](https://github.com/breezefront/module-breeze/blob/master/view/frontend/templates/catalog/recent-products.phtml) in your custom theme and render swatches manually using [Magento_Swatches/js/swatch-renderer widget](https://github.com/breezefront/module-breeze/blob/master/view/frontend/web/js/components/swatch-renderer.js#L122).
